CONTRIBUTING.md

```diff
@@ -0,0 +1,184 @@
+| help wanted      | Maintainers request community help resolving this issue.
+| Resolution (R)   | Indicates the resolution of an issue.
+| Status (S)       | Indicates the status of an issue.
+| Type (T)         | Indicates the type of an issue.
+
+## Submitting a Patch
+
+The Crust firmware project accepts patches through the use of GitHub pull
+requests. Although project maintainers accept impromptu pull requests on
+occasion, it is recommended that all pull requests correspond to an issue on
+the Crust firmware [issues page][issues]. Pull requests submitted _without_ a
+corresponding issue may be immediately rejected. If you believe Crust firmware
+could benefit from a new feature, please submit a feature request on the Crust
+firmware [issues page][issues].
+
+### Patch Lifecycle
+
+1. Comment on an issue from the Crust firmware [issues page][issues] to let
+   project maintainers know you would like to work on the issue. Project
+   maintainers will either acknowledge your comment or suggest another issue
+   if the issue is already being addressed.
+2. Submit a work-in-progress pull request (WIP) to receive early feedback from
+   project maintainers. Work-in-progress pull requests should be prefixed with
+   `WIP:` so project maintainers can assign labels to indicate the status.
+3. Address feedback from project maintainers. Prompt responses will allow a
+   pull request to get merged sooner!
+4. Complete the process until the pull request is approved by at least two
+   project maintainers.
+
+### Commits
+
+All contributions to this project must be released under the open-source
+license used by the project. To help ensure this, all commits you make must be
+accompanied by your real name and a Signed-off-by tag asserting authorship and
+permission to contribute, per the [Developer Certificate of Origin][dco]. This
+tag can be added automatically by git using `git commit -s`.
+
+Each change submitted in a pull request that addresses a different scope of the
+project should be contained within its own commit (with the exception of global
+changes, such as renaming files, that affect the entire scope of the project).
+
+For example, a pull request submitted for a new hardware component could
+include the following four commits:
+
+1. The first commit should add the headers and source files for the generic
+   driver class, if one is being added. (e.g. `drivers/<class>/<class>.c` and
+   `include/drivers/<class>.h`).
+2. The second commit should add the driver and its header
+   (e.g.  `drivers/<class>/<driver>.c` and `include/drivers/<class>/<driver>.h`).
+3. The third set of commits (one for each platform) should add the devices
+   that use this driver to `platform/<platform>/devices.c`. If this driver
+   supplies an additional platform-specific header, that should go in these
+   commits.
+4. Finally, the device should be utilized be used from common code (i.e. in
+   `common/main.c`).
```

Comment:
... or in `common/scpi_cmds.c` (or whatever we name it, since that's the second major "entry point" file)